backend/database/models/users.js

```diff
@@ -1,36 +1,105 @@
-    },
-    {
-      hooks: {
-        beforeCreate: User => {
-          const salt = bcrypt.genSaltSync();
-          User.password = bcrypt.hashSync(User.password, salt);
-          User.password_confirmation = User.password;
+    const users = sequelize.define(
+        'users',
+        {
+            username: {
+                type: DataTypes.STRING,
+                unique: true,
+            },
+            email: {
+                type: DataTypes.STRING,
+                validate: {
+                    isEmail: {
+                        msg: 'The email address you entered is invalid',
+                    },
+                },
+            },
+            password: DataTypes.STRING,
+            nickname: DataTypes.STRING,
+            profilePic: {
+                type: DataTypes.STRING,
+                validate: {
+                    isUrl: 'This image needs to be a link',
+                },
+            },
+            profileBanner: {
+                type: DataTypes.STRING,
+                validate: {
+                    isUrl: 'This image needs to be a link',
+                },
+            },
+            followers: {
+                type: DataTypes.STRING,
+                defaultValue: '[]',
+                get() {
+                    return JSON.parse(this.getDataValue('followers'))
+                },
+                set(val) {
+                    return this.setDataValue('followers', JSON.stringify(val))
+                },
+            },
+            following: {
+                type: DataTypes.STRING,
+                defaultValue: '[]',
+                get() {
+                    return JSON.parse(this.getDataValue('following'))
+                },
+                set(val) {
+                    return this.setDataValue('following', JSON.stringify(val))
+                },
+            },
+            posts: {
+                type: DataTypes.STRING,
+                defaultValue: '[]',
+                get() {
+                    return JSON.parse(this.getDataValue('posts'))
+                },
+                set(val) {
+                    return this.setDataValue('posts', JSON.stringify(val))
+                },
+            },
+            likes: {
+                type: DataTypes.STRING,
+                defaultValue: '[]',
+                get() {
+                    return JSON.parse(this.getDataValue('likes'))
+                },
+                set(val) {
+                    return this.setDataValue('likes', JSON.stringify(val))
+                },
+            },
+            shares: {
+                type: DataTypes.STRING,
+                defaultValue: '[]',
+                get() {
+                    return JSON.parse(this.getDataValue('shares'))
+                },
+                set(val) {
+                    return this.setDataValue('shares', JSON.stringify(val))
+                },
+            },
+            joinedDate: DataTypes.INTEGER,
+        },
```

Comment:
These fields need to be array of IDs, I think you can do that in sequelize by doing `DataTypes.ARRAY(DataTypes.INTEGER)`. Same thing needs to be done in create-user.js by doing `Sequelize.ARRAY(Sequelize.INTEGER)`

Reply:
(except joinedDate)

Reply:
Cool, I've made that change. I couldn't get ARRAY to work when I tested it. Although, I'm running mariadb not standard mysql